### PR TITLE
8328401: Convert java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.html applet test to automated

### PR DESCRIPTION
**Pull request by azvegint**
The test converted to automated, CI and manual testing looks good on all platforms.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328401](https://bugs.openjdk.org/browse/JDK-8328401): Convert java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.html applet test to automated (**Bug** - P4)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18367/head:pull/18367` \
`$ git checkout pull/18367`

Update a local copy of the PR: \
`$ git checkout pull/18367` \
`$ git pull https://git.openjdk.org/jdk.git pull/18367/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18367`

View PR using the GUI difftool: \
`$ git pr show -t 18367`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18367.diff">https://git.openjdk.org/jdk/pull/18367.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18367#issuecomment-2005343388)

### CODE REVIEW

File: test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java [hunk 1]
```diff
@@ -21,197 +21,82 @@
...
+        if (!Toolkit.getDefaultToolkit()
+                    .isFrameStateSupported(Frame.MAXIMIZED_BOTH)) {
+            return;
+        }
+
+        Robot robot = new Robot();
+        try {
+            EventQueue.invokeAndWait(InitialMaximizedTest::createAndShowFrame);
+            robot.waitForIdle();
+            robot.delay(2000);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java:55]: delay is too long ? Generally we have 1000 ms delay after creating test UI.

File: test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java [hunk 2]
```diff
@@ -21,197 +21,82 @@
...
+        Insets screenInsets = Toolkit.getDefaultToolkit().getScreenInsets(gc);
+        workArea.x += screenInsets.left;
+        workArea.y += screenInsets.top;
+        workArea.width -= screenInsets.left + screenInsets.right;
+        workArea.height -= screenInsets.top + screenInsets.bottom;
 
+        System.out.println("Frame bounds " + frameBounds);
+        System.out.println("GraphicsConfiguration bounds " + gc.getBounds());
+        System.out.println("Screen insets: " + screenInsets);
+        System.out.println("Work area: " + workArea);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java:81]: Is it intentional to keep so many print statement?

File: test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java [hunk 3]
```diff
@@ -21,197 +21,82 @@
...
-    public void displayMessage( String messageIn )
-    {
-        messageText.append( messageIn + "\n" );
-        System.out.println(messageIn);
+    private static void createAndShowFrame() {
+        frame = new Frame("The frame SHOULD be shown MAXIMIZED");
+        frame.setSize(300, 300);
+        frame.setLocation(50, 50);
+        frame.setExtendedState(Frame.MAXIMIZED_BOTH);
+        frame.addWindowListener(new WindowAdapter() {
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java:94]: Suggestion:
May be replaced with 
`frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);`

File: test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java (hunk 2 above)
Comment (azvegint) [reply test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java:81]: Looks cleaner to me than
```java
System.out.println(
        "Frame bounds " + frameBounds +
        "\nGraphicsConfiguration bounds " + gc.getBounds() +
        "\nScreen insets: " + screenInsets +
        "\nWork area: " + workArea
);
```

File: test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java (hunk 3 above)
Comment (azvegint) [reply test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java:94]: `java.awt.Frame` does not have the `setDefaultCloseOperation`, it is a `JFrame` method.

File: test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java (hunk 1 above)
Comment (azvegint) [reply test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java]: Reduced

File: test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java (hunk 3 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/Frame/InitialMaximizedTest/InitialMaximizedTest.java:94]: Ok. I misread it as JFrame.

**Review by kumarabhi006 — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**